types/react-maskedinput/index.d.ts

```diff
@@ -1,19 +1,48 @@
 import * as React from "react";
 
-export interface FormatCharacter {
+interface InternalFormatCharacter {
     validate(char: string): string;
     transform?(char: string): string;
 }
 
-export interface CharsFormatters {
-    [char: string]: FormatCharacter;
+interface InternalCharsFormatters {
+    [char: string]: InternalFormatCharacter;
 }
 
-export interface MaskedInputProps extends React.InputHTMLAttributes<any> {
+interface InternalMaskedInputProps extends
+    Omit<
+        React.InputHTMLAttributes<HTMLInputElement>,
+        | "maxLength"
+        | "onKeyDown"
+        | "onKeyPress"
+        | "onPaste"
+    >
+{
+    /**
+     * The masking pattern to be applied to the `<input>`.
+     * See the {@link https://github.com/insin/inputmask-core#pattern|inputmask-core} docs for
+     * supported formatting characters.
+     */
     mask: string;
-    formatCharacters?: CharsFormatters | undefined;
+    /**
+     * Customised format character definitions for use in the pattern.
+     * See the {@link https://github.com/insin/inputmask-core#formatcharacters|inputmask-core}
+     * docs for details of the structure of this object.
+     */
+    formatCharacters?: InternalCharsFormatters | undefined;
+    /**
+     * Customised placeholder character used to fill in editable parts of the pattern.
+     * See the {@link https://github.com/insin/inputmask-core#placeholderchar--string|inputmask-core}
+     * docs for details.
+     */
     placeholderChar?: string | undefined;
 }
 
-declare class MaskedInput extends React.Component<MaskedInputProps> {}
-export default MaskedInput;
+declare namespace MaskedInput {
+    type FormatCharacter = InternalFormatCharacter;
+    type CharsFormatters = InternalCharsFormatters;
+    type MaskedInputProps = InternalMaskedInputProps;
+}
```

Comment:
Why not declare them within the namespace?

Reply:
Moving the whole type body into namespace seems cannot help git to do comparison properly. 

I think the current approach make it more visible to reviewer (and to future maintainer) that, 
- which props typedef are actually changed, and 
- which props typedef remain the same with just additional jsdoc comment

Reply:
Viewing PRs without whitespace by adding `?w=1` is usually sufficient to fix that, e.g. https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70649/files?w=1

I would not use aliases for this. This technically represents a breaking change because type aliases cannot be merged with (e.g. a person cannot `declare module` to augment this library's interfaces, not that it's mega common).

Reply:
Noted. Thanks for the advice.